src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -64,10 +64,45 @@ public override BoundNode Visit(BoundNode node)
                 if (operation != null)
                 {
                     this.nodes.Add(operation);
+                    switch (operation.Kind)
+                    {
+                        case OperationKind.InvocationExpression:
+                            nodes.AddRange(((IInvocationExpression)operation).ArgumentsInSourceOrder);
+                            break;
+                        case OperationKind.ObjectCreationExpression:
+                            var objCreationExp = (IObjectCreationExpression)operation;
+                            nodes.AddRange(objCreationExp.ConstructorArguments);
+                            nodes.AddRange(objCreationExp.MemberInitializers);
+                            break;
+                    }
```

Comment:
It is still not clear what prevents us from adding duplicate nodes within this switch.